partner/partner-management-service/src/main/java/io/mosip/pms/partner/service/impl/PartnerServiceImpl.java

```diff
@@ -293,11 +293,15 @@ public PartnerResponse registerPartner(PartnerRequestDto request) {
 			isJSONValid(request.getAdditionalInfo().toString());
 		}
 
+		PolicyGroup policyGroupFromDb = policyGroupRepository.findByName(request.getPolicyGroup());
 		PartnerType partnerType = validateAndGetPartnerType(request.getPartnerType());
 		PolicyGroup policyGroup = null;
 		if (partnerType.getIsPolicyRequired()) {
 			policyGroup = validateAndGetPolicyGroupByName(request.getPolicyGroup());
 		}
+		else {
+			policyGroup=validatePolicyGroupByName(policyGroupFromDb)?policyGroupFromDb:null;
```

Comment:
validatePolicyGroupByName is returning Boolean, why are you setting it to PolicyGroup?